vignettes/SEIRDVAge.Rmd

```diff
@@ -0,0 +1,464 @@
+```
+
+Next, we set the parameter values for $\beta$, $\kappa$, $\gamma$, $\mu$, $\nu$,
+$\delta_V$, $\delta_R$ and $\delta_{VR}$:
+```{r}
+params <- list(beta=1, kappa=0.9, gamma=0.5, mu=0.1, nu=0.4, delta_V=0.1,
+               delta_R=0.05, delta_VR = 0.02)
+transmission_parameters(my_model) <- params
+```
+
+To check that the parameters have been set correctly, we call
+```{r}
+transmission_parameters(my_model)
+```
+
+We now need to set the initial conditions for the model. We only set the initial conditions $S(0)$, $E(0)$, $I(0)$, $R(0)$, $V(0)$ and $VR(0)$.
+```{r}
+initial_conditions(my_model) <- list(S0=pop_fraction*0.99,
+                                     E0=rep(0, n_ages),
+                                     I0=pop_fraction*0.01,
+                                     R0=rep(0, n_ages),
+                                     V0=rep(0, n_ages),
+                                     VR0=rep(0, n_ages),
+                                     D0=rep(0, n_ages))
+```
+The initial conditions must sum to one. If they do not, an error is thrown.
+
+Now we simulate the system from $t=0$ days to $t=200$ days.
+```{r}
+coverages <- vector(mode = "list", length = n_ages)
+
+for(i in 1:n_ages){
+  coverages[[i]] <- rep(0, 3*n_ages)
+  coverages[[i]][c(i:n_ages, (n_ages+i):(2*n_ages), (2*n_ages+i):(3*n_ages))] <- 1
+}
+
+intervention_parameters(my_model) <- list(starts=c(17+seq(1, n_ages), 50+seq(1, n_ages), 89+seq(1, n_ages)),
+                                          stops=c(c(18+seq(1, n_ages-1), 35), c(51+seq(1, n_ages-1), 69),
+                                                  c(90+seq(1, n_ages-1), 110)),
+                                          coverages=coverages)
+times <- seq(0, 200, by = 1)
+out_df <- run(my_model, times)
+```
+The added intervention feature allows for variable levels of vaccination in a population.
+In most cases, a vaccine is not available or considered necessary right from the
+start of an epidemic so there is a delay in vaccination. Similarly, vaccination
+programs may not run at the same rate all the time, so such fluctuations need to
+be taken into account in our modeling. Vaccination levels and start times may differ between
+different age groups. In this instantiation of the model, vaccination of the each age group occurs
+at a different stage
+
+Below we can see a plot of the a claassical level of interventions that occur at each time step.
+```{r, fig.height = 5, fig.width = 7}
+int_parms <- InterventionParameters(start=c(17, 35, 42),
+                                    stop=c(25, 39, 49),
+                                    coverage=c(1, 1, 1))
+
+sim_parms <- SimulationParameters(start=0, stop=200, tstep = 0.1)
+
+intervention_protocol(int_parms, sim_parms, 1) %>%
+  ggplot(aes(x=time, y=coverage)) +
+  geom_line() +
+  scale_color_brewer(palette = "Dark2") +
+  labs(x = "time (days)", y = "level of intervention") +
+  theme(text = element_text(size = 20))
```

Comment:
It would be better to here plot the actual age structured intervention, either with different colors for each age group or one panel for each group's intervention